client/app/deploy/services/serverRoleService.js

```diff
@@ -0,0 +1,149 @@
+/* TODO: enable linting and fix resulting errors */
+/* eslint-disable */
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+
+'use strict';
+
+(function () {
+  angular
+    .module('EnvironmentManager.deploy')
+    .factory('clientServerRoleService', serverRoleService);
+
+  serverRoleService.$inject = ['$http'];
+
+  function serverRoleService($http) {
+    return {
+      create: create
+    };
+
+    function reduceDeploymentMaps(model) {
+      return model.DeploymentMaps.reduce(function (acc, curr) {
+        acc[curr.SelectedName] = acc[curr.SelectedName] || [];
+        acc[curr.SelectedName].push(curr);
+        return acc;
+      }, {});
+    }
```

Comment:
Recreation of groupby... `return _.groupBy(sampleMaps, 'SelectedName');`

Reply:
Yup worked!